norfair/drawing.py

```diff
@@ -449,3 +461,67 @@ def random(obj_id: int) -> Tuple[int, int, int]:
             and c not in ("random", "red", "white", "grey", "black", "silver")
         ]
         return getattr(Color, color_list[obj_id % len(color_list)])
+
+
+from functools import lru_cache
+
+
+@lru_cache(maxsize=4)
+def _get_grid(size, w, h, polar=False):
+    """
+    Construct the grid of points.
+
+    Points are choosen
+    Results are cached since the grid in absolute coordinates doesn't change.
+    """
+    # We need to get points on a semi-sphere of radious 1 centered around (0, 0)
+
+    # First step is to get a grid of angles, theta and phi ∈ (-pi/2, pi/2)
+    step = np.pi / size
+    start = -np.pi / 2 + step
+    end = np.pi / 2 - step
+    theta, fi = np.mgrid[start:end:step, start:end:step]
```

Comment:
I would suggest (only to have the whole circle equispaced) to move the start `step / 2` and set the `end` value to just `np.pi / 2` (when creating the grid, it will use use `np.pi / 2 - step / 2` as the maximum value anyways because the start is shifted half a step), as in:

```
step = np.pi / size
start = -np.pi / 2 + step / 2
end = np.pi / 2
theta, fi = np.mgrid[start:end:step, start:end:step]
```